### PR TITLE
Fix api

### PR DESCRIPTION
**Pull request by alexyao2015**
Fixes issue where RGBWW lights didn't show both the color wheel and temp wheel. Various tweaks

### CODE REVIEW

**Review by marcelveldt — CHANGES_REQUESTED** [review]
the import is in the wrong order

File: emulated_hue/api.py [hunk 1]
```diff
@@ -663,6 +669,12 @@ async def __async_entity_to_hue(
                     const.HUE_ATTR_XY: entity_attr.get(
                         const.HASS_ATTR_XY_COLOR, [0, 0]
                     ),
+                    const.HUE_ATTR_HUE: entity_attr.get(
```
Comment (marcelveldt) [inline emulated_hue/api.py:686]: I'm not so sure if setting all the values at once is a good idea. It's either xy, hs or ct not all at the same time.
We'll have to see if this doesn't bring us in trouble with some testing.

File: emulated_hue/api.py [hunk 2]
```diff
@@ -436,9 +438,23 @@ async def async_change_config(self, request: web.Request, request_data: dict):
...
         LOGGER.debug("Change config called with params: %s", request_data)
         for key, value in request_data.items():
-            await self.config.async_set_storage_value("bridge_config", key, value)
+            if key == "linkbutton" and value and not self.config.link_mode_enabled:
+                await self.config.async_enable_link_mode()
+            else:
+                await self.config.async_set_storage_value("bridge_config", key, value)
         return send_success_response(request.path, request_data, username)
 
+    async def async_scene_to_full_state(self) -> dict:
```
Comment (marcelveldt) [inline emulated_hue/api.py:447]: Will this also support scenes where the lights have different settings ?

**Review by marcelveldt — APPROVED** [review]
Let's merge this in and do some testing!

File: emulated_hue/api.py [hunk 3]
```diff
@@ -9,6 +10,7 @@
 from typing import Any, AsyncGenerator, Optional
 
 import emulated_hue.const as const
```
Comment (marcelveldt) [inline emulated_hue/api.py:12]: move the emulated_hue.const import

use isort to auto fix it.

File: emulated_hue/api.py (hunk 1 above)
Comment (alexyao2015) [reply emulated_hue/api.py:686]: Its actually required by the hue apps and is how the official bridge responds. Fixes the hue app not showing both color modes.

File: emulated_hue/api.py (hunk 2 above)
Comment (alexyao2015) [reply emulated_hue/api.py:447]: What do you mean by different settings? All this does is delete the lightstates and add the lights in the group. Do you have a better way of implementing this? I'm not very satisfied with how I've implemented it, but it does work.

**Comment by alexyao2015** [comment]
> the import is in the wrong order

Which import?

File: emulated_hue/api.py (hunk 2 above)
Comment (marcelveldt) [reply emulated_hue/api.py:447]: Never mind. I'll give it a test once merged.

File: emulated_hue/api.py (hunk 3 above)
Comment (alexyao2015) [reply emulated_hue/api.py:12]: Believe it or not, precommit moved it there. I used pycharm to sort it and it moved it to the way you would think, but precommit decided to move it back to there.

Comment (marcelveldt) [reply emulated_hue/api.py:12]: hmmm strange. maybe some setting in isort (which is used by pre commit) ?

**Pull request merged**